core/src/main/scala/org/apache/spark/storage/BlockManagerMasterEndpoint.scala

```diff
@@ -336,13 +345,16 @@ class BlockManagerMasterEndpoint(
   private def removeBlockManager(blockManagerId: BlockManagerId): Unit = {
     val info = blockManagerInfo(blockManagerId)
 
+    // Not removing info from the blockManagerInfo map, but only updating the removal timestamp of
```

Comment:
nit: "updating" -> "setting"

(I think it'd be only removed once.)

Reply:
Done.

Reply:
Could you also explain the issue with removing here?

Reply:
Done.

Reply:
Please let me know if any more changes are required. 😄